plugins/toolbox/src/components/DefaultEditor/TripleEditor.tsx

```diff
@@ -0,0 +1,271 @@
+
+type Props = {
+  input: string;
+  setInput: (value: string) => void;
+  pattern: string;
+  setPattern: (value: string) => void;
+  output?: string;
+  setOutput: (value: string) => void;
+  mode?: string;
+  minRows?: number;
+  inputLabel?: string;
+  patternLabel?: string;
+  outputLabel?: string;
+  setMode?: (value: string) => void;
+  modes?: Array<string>;
+  leftContent?: ReactElement;
+  extraLeftContent?: ReactElement;
+  middleContent?: ReactElement;
+  extraMiddleContent?: ReactElement;
+  rightContent?: ReactElement;
+  extraRightContent?: ReactElement;
+  sample?: string;
+  additionalTools?: ReactElement[];
+  allowFileUpload?: boolean;
+  acceptFileTypes?: string;
+  allowFileDownload?: boolean;
+  downloadFileType?: string;
+  downloadFileName?: string;
+  inputProps?: any;
+  outputProps?: any;
+};
+
+export const TripleEditor = (props: Props) => {
+  const {
+    input,
+    setInput,
+    pattern,
+    setPattern,
+    output,
+    setOutput,
+    inputLabel = 'Input',
+    patternLabel = 'Pattern',
+    outputLabel = 'Output',
+    mode,
+    setMode,
+    modes,
+    leftContent,
+    extraLeftContent,
+    middleContent,
+    extraMiddleContent,
+    rightContent,
+    extraRightContent,
+    sample,
+    additionalTools,
+    allowFileUpload,
+    acceptFileTypes,
+    allowFileDownload,
+    downloadFileName,
+    downloadFileType,
+    minRows = 20,
+  } = props;
+  const { classes } = useStyles();
+
+  const [fileName, setFileName] = React.useState(
+    downloadFileName ?? 'download.txt',
+  );
+  const [fileType, setFileType] = React.useState(
+    downloadFileType ?? 'text/plain',
+  );
+
+  const readFileAndSetInput = (file?: File) => {
+    if (!file) {
+      setInput('');
+      return;
+    }
+
+    setFileName(file.name);
+    setFileType(file.type);
+    const reader = new FileReader();
+    reader.onload = async e => {
+      // @ts-ignore
+      setInput(e.target.result);
+    };
+    reader.readAsText(file);
+  };
+
+  const handleDrop = (ev: React.DragEvent<HTMLDivElement>) => {
+    if (allowFileUpload !== true) {
+      return;
+    }
+    ev.preventDefault();
+    if (ev.dataTransfer.items) {
+      [...ev.dataTransfer.items].forEach(item => {
+        if (item.kind !== 'file') {
+          return;
+        }
+        const file = item.getAsFile();
+        if (file) {
+          readFileAndSetInput(file);
+        }
+      });
+    } else {
+      [...ev.dataTransfer.files].forEach(file => {
+        readFileAndSetInput(file);
+      });
+    }
+  };
+
+  return (
+    <FormControl className={classes.fullWidth} onDrop={handleDrop}>
+      <Grid container spacing={4} style={{ marginBottom: '5px' }}>
+        {modes && modes.length > 0 && (
+          <Grid item sx={{ pl: '16px', pt: '32px !important' }}>
+            <ButtonGroup
+              size="small"
+              disableElevation
+              variant="contained"
+              aria-label="Disabled elevation buttons"
```

Comment:
Translations in use here

Reply:
translated

Reply:
Hm, you should use the `const { t } = useToolboxTranslation();` for this as well